### PR TITLE
Implement vector3 with common widget vector3

### PR DESCRIPTION
**Pull request by AzulRadio**
Signed-off-by: youhy <haoyuan2019@outlook.com>

* Part of https://github.com/gazebosim/gz-gui/issues/310

# New feature

## Summary
Implement `Gravity` and `Magnetic Field` in Component Inspector with common widget GzVector3 introduced in gz-gui#427

Behavior should be exactly the same as before.

image

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/gui/plugins/component_inspector/Vector3d.qml [hunk 1]
```diff
@@ -35,46 +35,9 @@ Rectangle {
   // Horizontal margins
   property int margin: 5
 
-  // Maximum spinbox value
-  property double spinMax: 1000000
-
   // Units, defaults to meters.
   property string unit: model && model.unit != undefined ? model.unit : 'm'
```
Comment (jennuine) [inline src/gui/plugins/component_inspector/Vector3d.qml:39]: Let's remove this and move it below to where `gzUnit` is. Eg.
```javascript
gzUnit: model && model.unit != undefined ? model.unit : 'm'
```

File: src/gui/plugins/component_inspector/Vector3d.qml [hunk 2]
```diff
@@ -25,7 +24,8 @@ import "qrc:/qml"
 
 // Item displaying 3D vector information.
 Rectangle {
-  height: header.height + content.height
+
```
Comment (jennuine) [inline src/gui/plugins/component_inspector/Vector3d.qml:27]:
```suggestion
```

**Review by jennuine — APPROVED** [review]

File: src/gui/plugins/component_inspector/Vector3d.qml (hunk 2 above)
Comment (AzulRadio) [reply src/gui/plugins/component_inspector/Vector3d.qml]: Fixed in a8c5a7d

File: src/gui/plugins/component_inspector/Vector3d.qml (hunk 1 above)
Comment (AzulRadio) [reply src/gui/plugins/component_inspector/Vector3d.qml]: Fixed in a8c5a7d

**Pull request merged**